lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -199,6 +213,53 @@ fetchDelistedPools tr uri manager = runExceptTLog $ do
                 , show s
                 ]
 
+    fromHttpException :: Monad m => HttpException -> m (Either String a)
+    fromHttpException = return . Left . ("HTTP exception: " <>) . show
+
+-- | Gets the health status from the SMASH server. Returns
+-- @Nothing@ if the server is unreachable.
+healthCheck
+    :: Maybe (Tracer IO StakePoolMetadataFetchLog)
+    -> URI
+    -> Manager
+    -> IO (Maybe HealthStatusSMASH)
+healthCheck mtr uri manager = runExceptTLog $ do
+    pl <- smashRequest mtr
+        (uri { uriPath = "/" <> healthCheckEP , uriQuery = "", uriFragment = "" })
+        manager
+    except . eitherDecodeStrict @HealthStatusSMASH $ pl
+  where
+    runExceptTLog
```

Comment:
Failure here is HTTP failure, success is just "we got an answer" and doesn't mean the server is healthy. I'd like to keep those separate, because they currently don't have the same severity either.

Reply:
But you are right in the sense that the response object shouldn't have information that diverges from the HTTP status code it returns.

Reply:
Whatever "result" type you choose, it can be pattern matched in the `getSeverityAnnotation` definition.